### PR TITLE
CI Test

### PR DESCRIPTION
**Pull request by zhhsplendid**
### PR types

New features
### PR changes

Others
### Describe

1. Add piano_ci_test.sh, which runs the ctests that has corresponding files under `paddle/fluid/compiler` folder.
2. Add `libtinfo-dev` to Dockerfile
3. Change `piano_layout_test` and `piano_shape_test` to `layout_test` and `shape_test`
4. Add `-DWITH_COMPILER` into `paddle_build.sh`

### CODE REVIEW

File: paddle/fluid/compiler/piano/CMakeLists.txt [hunk 1]
```diff
@@ -2,5 +2,5 @@ add_subdirectory(backends)
 add_subdirectory(note)
 
 cc_library(piano_data_description SRCS layout.cc shape.cc DEPS string_helper note_proto)
-cc_test(piano_layout_test SRCS layout_test.cc DEPS piano_data_description)
-cc_test(piano_shape_test SRCS shape_test.cc DEPS piano_data_description)
+cc_test(layout_test SRCS layout_test.cc DEPS piano_data_description)
+cc_test(shape_test SRCS shape_test.cc DEPS piano_data_description)
```
Comment (wzzju) [inline paddle/fluid/compiler/piano/CMakeLists.txt:6]: Just a guess: Without `piano_` prefix, they may conflict with other unit tests.

File: paddle/fluid/compiler/scripts/piano_ci_test.sh [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+# limitations under the License.
+
+set -ex
+
+if [ -z ${BRANCH} ]; then
+    BRANCH="paddle_compiler"
+fi
+
+function init() {
+    RED='\033[0;31m'
```
Comment (thisjiang) [inline paddle/fluid/compiler/scripts/piano_ci_test.sh:24]: 这些颜色值是不是没有用到啊？另外`\033[`开头的ANSI转义代码在windows cmd和powershell上print会乱码，之前有过解决过类似问题： https://github.com/PaddlePaddle/Paddle/pull/33689

Comment (zhhsplendid) [reply paddle/fluid/compiler/scripts/piano_ci_test.sh:24]: 好的，后续PR删除掉

File: paddle/fluid/compiler/piano/CMakeLists.txt (hunk 1 above)
Comment (zhhsplendid) [reply paddle/fluid/compiler/piano/CMakeLists.txt:6]: Yes, check your group chat offline

**Review by wzzju — APPROVED** [review]
LGTM

**Review by Xreki — APPROVED** [review]

**Pull request merged**